src/main/ArgsParser.java

```diff
@@ -0,0 +1,80 @@
+package main;
+
+import main.algorithms.Algorithm;
+import main.algorithms.ShiftAlgorithmImpl;
+import main.algorithms.UnicodeAlgorithmImpl;
+
+public class ArgsParser {
+
+    private String mode;
+    private String data;
+    private int key;
+    private String in;
+    private String out;
+    private Algorithm algorithm;
+
+    public ArgsParser() {
+        mode = "enc";
+        data = "";
+        key = 0;
+        out = "";
+    }
+
+    public void parseArgs(String[] args) {
+        for (int i = 0; i < args.length; i += 2) {
+            switch (args[i]) {
+                case "-mode" -> mode = args[i + 1];
+                case "-key" -> {
+                    try {
+                        key = Integer.parseInt(args[i + 1]);
+                    } catch (NumberFormatException e) {
+                        System.err.println("Wrong offset (key) format");
+                        System.exit(1);
+                    }
+                }
+                case "-data" -> data = args[i + 1];
+                case "-in" -> in = args[i + 1];
+                case "-out" -> out = args[i + 1];
+                case "-alg" -> {
+                    if (args[i + 1].equals("unicode")) {
```

Comment:
The "-alg" may never come and in this case, you will get NPE in the result.

Reply:
If you leave this piece of code it will be better to move it to a separate method.

Reply:
Can you make this string "unicode" a constant with a meaningful name?

Reply:
fixed NPE, extracted parsing algorithm argument to another method and made an Enum with labels of each algorithm.